app/controllers/posts_controller.rb

```diff
@@ -120,10 +113,22 @@ def check_if_needs_redirect
       end
     end
 
+    def fetch_purchase
+      @purchase = current_seller.sales.find_by_external_id(params[:purchase_id])
+      return e404_json if @purchase.blank?
+
+      @seller = @purchase.seller
```

Comment:
Setting `@seller` in purchase as well to use it in `ensure_seller_is_eligible_to_send_emails` down below. 

This ensures we have always have a seller available before we make an eligibility check